panic: invalid argument to Intn goroutine

Version 0.9
Druid version 0.20.0

```
panic: invalid argument to Intn goroutine 23214 [running]:
math/rand.(*Rand).Intn(0xc00009e060, 0x0, 0xc00115e000)
	/usr/local/go/src/math/rand/rand.go:169 +0x9c
math/rand.Intn(...)
	/usr/local/go/src/math/rand/rand.go:337
druid-exporter/collector.(*MetricCollector).Collect(0xc000376140, 0xc0000ad200)
	/go/src/druid-exporter/collector/druid.go:171 +0xad2
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	/go/pkg/mod/github.com/prometheus/client_golang@v1.5.1/prometheus/registry.go:443 +0x19d
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/go/pkg/mod/github.com/prometheus/client_golang@v1.5.1/prometheus/registry.go:454 +0x599
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,16 @@
+### v0.9
+##### January 13, 2020
+
+#### :beetle: Bug Fixes
+
+- Fixed interface to a nil conversion error in Druid HTTP Endpoint
+- Invalid argument(0) to Intn goroutine
+
+#### :tada: Features
+
+- Added datasource row count as metrics
+- Docker compose setup to fix
+
 ### v0.8
 ##### July 11, 2020
 

--- a/collector/druid.go
+++ b/collector/druid.go
@@ -189,12 +189,18 @@ func (collector *MetricCollector) Collect(ch chan<- prometheus.Metric) {
 	for _, data := range GetDruidSegmentData() {
 		ch <- prometheus.MustNewConstMetric(collector.DataSourceCount,
 			prometheus.GaugeValue, float64(1), data.Name)
-		ch <- prometheus.MustNewConstMetric(collector.DruidSegmentCount,
-			prometheus.GaugeValue, float64(data.Properties.Segments.Count), data.Name)
-		ch <- prometheus.MustNewConstMetric(collector.DruidSegmentSize,
-			prometheus.GaugeValue, float64(data.Properties.Segments.Size), data.Name)
-		ch <- prometheus.MustNewConstMetric(collector.DruidSegmentReplicateSize,
-			prometheus.GaugeValue, float64(data.Properties.Segments.ReplicatedSize), data.Name)
+		if data.Properties.Segments.Count != 0 {
+			ch <- prometheus.MustNewConstMetric(collector.DruidSegmentCount,
+				prometheus.GaugeValue, float64(data.Properties.Segments.Count), data.Name)
+		}
+		if data.Properties.Segments.Size != 0 {
+			ch <- prometheus.MustNewConstMetric(collector.DruidSegmentSize,
+				prometheus.GaugeValue, float64(data.Properties.Segments.Size), data.Name)
+		}
+		if data.Properties.Segments.ReplicatedSize != 0 {
+			ch <- prometheus.MustNewConstMetric(collector.DruidSegmentReplicateSize,
+				prometheus.GaugeValue, float64(data.Properties.Segments.ReplicatedSize), data.Name)
+		}
 	}
 
 	workers := getDruidWorkersData(workersURL)

